frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -349,6 +351,39 @@ impl<T: Config> SignedSubmissions<T> {
 	}
 }
 
+/// Type that can be used to calculate the deposit base for signed submissions.
+///
+/// The deposit base is calculated as a geometric progression based on the number of signed
+/// submissions in the queue. The size of the queue represents the progression term.
+pub struct GeometricDepositBase<T> {
+	_marker: PhantomData<T>,
+}
+
+impl<T: Config> DepositCalculator<BalanceOf<T>> for GeometricDepositBase<T> {
+	/// The fixed deposit base upon which the geometric series is calculated.
+	type BaseDeposit = T::SignedFixedDepositBase;
+
+	/// Increase factor of the geometric progression, as a percentage.
+	///
+	/// The nth progression term will be `IncreaseFactor`% larger than the nth-1 term.
+	type IncreaseFactor = T::SignedDepositBaseIncreaseFactor;
+
+	/// Increase the base deposit as a geometric progression based on the number of signed
```

Comment:
nit: implementations can be commented with just `//`, not `///`. See https://hackmd.io/mRrrewoxTOyfKJ2H16R2xw?both, to be made public soon.